service/src/main/java/org/mskcc/smile/service/impl/RequestServiceImpl.java

```diff
@@ -172,17 +172,26 @@ public PublishedSmileRequest getPublishedSmileRequestById(String requestId) thro
     }
 
     @Override
-    public Boolean updateRequestMetadata(RequestMetadata requestMetadata) throws Exception {
+    public Boolean updateRequestMetadata(RequestMetadata requestMetadata, Boolean fromLims) throws Exception {
         SmileRequest existingRequest = getSmileRequestById(requestMetadata.getIgoRequestId());
         if (existingRequest == null) {
             LOG.error("Cannot persist updates to a request that does not already exist: "
                     + requestMetadata.getIgoRequestId());
             return Boolean.FALSE;
         }
-        // persist updates for request metadata if applicable
-        if (requestHasMetadataUpdates(existingRequest.getLatestRequestMetadata(), requestMetadata)) {
+        // persist igo updates for request metadata if applicable
+        if (fromLims && requestHasMetadataUpdates(existingRequest.getLatestRequestMetadata(),
+                requestMetadata, Boolean.TRUE)) {
             LOG.info("Persisting updates for request: " + existingRequest.getIgoRequestId());
-            existingRequest.updateRequestMetadataByMetadata(requestMetadata);
+            existingRequest.applyIgoRequestMetadataUpdates(requestMetadata);
+            saveRequestMetadata(existingRequest);
+            return Boolean.TRUE;
+        }
+        // persist dashboard/user updates for request metadata if applicable
+        if (!fromLims && requestHasMetadataUpdates(existingRequest.getLatestRequestMetadata(),
+                requestMetadata, Boolean.FALSE)) {
+            LOG.info("Persisting updates for request: " + existingRequest.getIgoRequestId());
+            existingRequest.applyDashboardRequestMetadataUpdates(requestMetadata);
             saveRequestMetadata(existingRequest);
             return Boolean.TRUE;
```

Comment:
this bit is a little confusing to me because we have the `fromLims` flag here but then we're also setting it to true in this method call? `requestHasMetadataUpdates(existingRequest.getLatestRequestMetadata(),
                requestMetadata, Boolean.TRUE)`? 

i guess i see the differences between these two blocks - one applies igo request updates and the other applies dashboard updates. I still don't think that the dashboard fields are something we should explicitly define but maybe we can talk about it more with @n1zea144 as well?